src/bootstrap/dist.rs

```diff
@@ -439,51 +440,52 @@ pub fn rust_src(build: &Build) {
         copy(&build.src.join(item), &plain_dst_src.join(item));
     }
 
-    // If we're building from git sources, we need to vendor a complete distribution.
-    if build.src_is_git {
-        // Get cargo-vendor installed, if it isn't already.
-        let mut has_cargo_vendor = false;
-        let mut cmd = Command::new(&build.cargo);
-        for line in output(cmd.arg("install").arg("--list")).lines() {
-            has_cargo_vendor |= line.starts_with("cargo-vendor ");
-        }
-        if !has_cargo_vendor {
+    // Create the version file
+    write_file(&plain_dst_src.join("version"), build.rust_version().as_bytes());
+
+    if let Subcommand::Dist { .. } = build.flags.cmd {
```

Comment:
Yes. This part generates both the rust-src component installer and the full source tarball with vendorized deps, etc, and we only want the former for install